src/Instrumentation/MySqli/src/MySqliInstrumentation.php

```diff
@@ -0,0 +1,832 @@
+    {
+        if ($retVal != true) {
+            return;
+        }
+        $tracker->addMySqliAttribute($obj ? $obj : $params[0], TraceAttributes::DB_NAMESPACE, $params[$obj ? 0 : 1]);
+    }
+
+    /** @param non-empty-string $spanName */
+    private static function preparePreHook(string $spanName, CachedInstrumentation $instrumentation, MySqliTracker $tracker, $obj, array $params, ?string $class, ?string $function, ?string $filename, ?int $lineno): void
+    {
+        $span = self::startSpan($spanName, $instrumentation, $class, $function, $filename, $lineno, []);
+        $mysqli = $obj ? $obj : $params[0];
+        self::addTransactionLink($tracker, $span, $mysqli);
+    }
+
+    private static function preparePostHook(CachedInstrumentation $instrumentation, MySqliTracker $tracker, $obj, array $params, mixed $stmtRetVal, ?\Throwable $exception)
+    {
+        $mysqli = $obj ? $obj : $params[0];
+        $query = $params[$obj ? 0 : 1];
+
+        $errorStatus = null;
+
+        $query = mb_convert_encoding($query, 'UTF-8');
+        $operation = self::extractQueryCommand($query);
+
+        $attributes = $tracker->getMySqliAttributes($mysqli);
+        $attributes[TraceAttributes::DB_STATEMENT] = $query;
+        $attributes[TraceAttributes::DB_OPERATION_NAME] = $operation;
+
+        if (!$exception && $stmtRetVal instanceof mysqli_stmt) {
+            $tracker->trackMySqliFromStatement($mysqli, $stmtRetVal);
+            $tracker->addStatementAttribute($stmtRetVal, TraceAttributes::DB_STATEMENT, $query);
+            $tracker->addStatementAttribute($stmtRetVal, TraceAttributes::DB_OPERATION_NAME, $operation);
+
+        } else {
+            //TODO use constant from comment  after sem-conv update
+            $attributes[/*TraceAttributes::DB_RESPONSE_STATUS_CODE*/ 'db.response.status_code'] =  $mysqli->errno;
+            $errorStatus = !$exception ? $mysqli->error : null;
+        }
+
+        self::endSpan($attributes, $exception, $errorStatus);
+    }
+
+    /** @param non-empty-string $spanName */
+    private static function beginTransactionPreHook(string $spanName, CachedInstrumentation $instrumentation, MySqliTracker $tracker, $obj, array $params, ?string $class, ?string $function, ?string $filename, ?int $lineno): void
+    {
+        self::startSpan($spanName, $instrumentation, $class, $function, $filename, $lineno, []);
+    }
+
+    private static function beginTransactionPostHook(CachedInstrumentation $instrumentation, MySqliTracker $tracker, $obj, array $params, mixed $retVal, ?\Throwable $exception)
+    {
+        $mysqli = $obj ? $obj : $params[0];
+        $transactionName = $params[$obj ? 1 : 2] ?? null;
+
+        $attributes = $tracker->getMySqliAttributes($mysqli);
+
+        if ($transactionName) {
+            $attributes['db.transaction.name'] = $transactionName;
```

Comment:
In reference to the discussion on SIG and the custom attribute db.transaction.name, I looked into it and found that there is no corresponding entity in sem-conv. I started digging deeper into the topic and discovered that this is typical for the MySQLi extension in PHP. PDO does not allow setting a transaction name.

I checked the code of mysqlnd (the base driver for MySQLi and PDO), and it seems that there is no such concept as a transaction name. In mysqlnd, this is implemented by setting a comment in the SQL code—it is not interpreted by the database itself as a name but is treated as a regular comment.